web/src/features/charts/bar-breakdown/ProductionSourceLegend.tsx

```diff
@@ -0,0 +1,31 @@
+import { ElectricityModeType } from 'types';
+import { modeColor } from 'utils/constants';
+
+import { iconHeight, LABEL_MAX_WIDTH, PADDING_Y } from './constants';
+import { getIconPadding } from './utils';
+
+export default function ProductionSourceLegend({
+  electricityType,
+}: {
+  electricityType: ElectricityModeType;
+}) {
+  return (
+    <g>
+      <rect
+        transform={`translate(${LABEL_MAX_WIDTH - 1.5 * PADDING_Y - 10}, 0)`}
+        fill={modeColor[electricityType as ElectricityModeType]}
+        width={14}
+        height={14}
+        rx={2}
+      />
+      <image
+        transform={`translate(${LABEL_MAX_WIDTH - 1.5 * PADDING_Y - 10}, ${getIconPadding(
+          electricityType
+        )})`}
+        width={14}
+        height={iconHeight[electricityType as ElectricityModeType]}
+        xlinkHref={`/images/production-source/${electricityType}.svg`}
+      />
+    </g>
+  );
```

Comment:
This is mixing svg elements and the image, I think it would be better to just have a `div` instead of `rect` if possible :)

Reply:
Following up on this, I think it would make even more sense to import the svgs directly using a dynamic import and style it instead of an image. By importing the SVG vite should be able to do some bundle magic and include it in our bundle to avoid HTTP calls.

Reply:
Aaah wait, now I understand why you're using the `rect` though, didn't realise it was all within a big SVG! Then it makes good sense...

Agree on Viktor's comment though :)

Reply:
I guess it would also be possible to make them into components as svg code is also valid html if that is easier.